src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -1029,9 +1009,7 @@ private static int LocateLastFoundChar(Vector<ushort> match)
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int LocateLastFoundChar(ulong match)
-        {
-            return 3 - (BitOperations.LeadingZeroCount(match) >> 4);
-        }
+            => BitOperations.Log2Unsafe(match) >> 4;
```

Comment:
In this use, `Log2` is not necessarily more expressive of the intent, but it's more expressive of the end result.  This is calculating Log<sub>2</sub><sup>16</sup>, for the purpose of finding a 16-bit character position.  The `LeadingZeroCount` version could be improved by replacing subtraction with xor, but this is cleaner.

Again, the `Log2` adds an extra unnecessary instruction, but I was unable to measure a difference in benchmarks.  `Log2Unsafe` simply eliminates the unnecessary `or`.

Reply:
I don't think a single `or` instruction, which takes a single cycle, is that much of a concern. It would be better to just use `Log2` and burn the quarter nanosecond.

Reply:
I would agree that a single instruction is not a huge concern.  However, if eliminating it is 100% safe and does not complicate the code, then why not.  In fact, I'd argue that the normal `Log2` -- which does extra work to guarantee a result for an undefined operation -- is the one that's out of place.  But since the API is public and the result for log(0) is guaranteed, it must remain so.

Clearly there's no appetite for `Log2Unsafe` among the team, so I will back it out.  My main concern was that I not regress in a perf-sensitive area of the code, but the regression here is compensated for by the elimination of the subtraction from a constant.  So even with the redundant `or`, this specific change is a wash for `lzcnt`-capabale environments and a gain for those that would use the a log2 fallback to emulate `lzcnt`.

Reply:
I think the argument is that is does complicate the code. Someone reading `BitOperations.Log2()` or `BitOperations.LeadingZeroCount()` will know immediately what it is doing. While someone reading `BitOperations.Log2Unsafe()` now has to consider what is `Unsafe` about it and what special considerations need to be given to this code. It immediately raises the bar and complexity level for anyone coming into this code in the future.

Reply:
Fair enough. I can't say I'm a fan of the `Unsafe` paradigm as it's used here and in other places, but this seems to be the pattern.  `Log2OfNonZeroValue` would make it the usage clear on first view, but it's an ugly name.  Not a big deal either way, though.

Reply:
I've reverted the `Log2Unsafe` addition. It occurred to me it might have been better to create a `FindFirstSet` and `FindLastSet` pair since we rarely care how many leading/trailing 0's there are but rather where the first/last 1 is.  Would still be nice to get rid of the branches in the BSF/BSR fallbacks that emulate TZCNT/LZCNT when the argument is known to be non-zero, but I'll leave that alone for now.